File: apps/mobile/lib/models/audit.dart
class AuditLog {
  final String id;
  final String eventType;
  final String entityType;
  final String? entityId;
  final String? user;
  final Map<String, dynamic>? details;
  final DateTime createdAt;

  const AuditLog({
    required this.id,
    required this.eventType,
    required this.entityType,
    this.entityId,
    this.user,
    this.details,
    required this.createdAt,
  });

  factory AuditLog.fromJson(Map<String, dynamic> json) {
    return AuditLog(
      id: json['id'] as String,
      eventType: json['event_type'] as String,
      entityType: json['entity_type'] as String,
      entityId: json['entity_id'] as String?,
      user: json['user'] as String?,
      details: json['details'] != null
          ? Map<String, dynamic>.from(json['details'])
          : null,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}

File: crates/core/src/models/audit.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use serde_json::Value;
use uuid::Uuid;

use crate::types::AuditEventType;

/// A single audit log entry.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AuditLog {
    pub id: Uuid,
    pub event_type: AuditEventType,
    pub entity_type: String,
    pub entity_id: Option<Uuid>,
    pub user: Option<String>,
    pub details: Option<Value>,
    pub created_at: DateTime<Utc>,
}
